### PR TITLE
Changed when the game sends classify request to backend

### PR DESCRIPTION
**Pull request by andreasrberg**
The game now uses more complex logic to decide when to classify the drawing. The baseline is as follows:

- The game starts guessing words after x amount of seconds, which differs based on difficulty
- The game will always guess x amount of seconds after the last guess (every 4 or 5 seconds, based on difficulty)
- The game will not start guessing before a minimum threshold of pixels has been drawn on the screen
- If the user has drawn x amount of pixels since the last guess, the game will guess again as long as more than 2 seconds has elapsed since the last classification
- The game will additionally not make guesses while the player is drawing (So it will instantly classify if the user lets go of the touch screen/mouse)

This can be seen in the game-config.service.ts where the following attributes has been added:
- defaultTimeBetweenClassify 
- minimumTimeBetweenClasssify
- MinimumDrawnThreshold
- pixelsPerClassify 

The conditions checking if the game should classify is placed into the `checkClassificationConditions()` function in game-draw.component.ts

Additionally, the game checks how much is drawn on the screen through the new `countDrawnPixels()` function in the same component.

The specific values in the game config could be altered further based on user testing, though the feel of the game is still mostly the same

### CODE REVIEW

File: src/app/game/game-draw/game-draw.component.ts [hunk 1]
```diff
@@ -1,468 +1,528 @@
...
+    const hasPassedMinimumDrawnThreshold = this.drawnPixels > this.config.minimumDrawnThreshold;
+    const hasDrawnSinceLastClassify = this.drawnPixelsSinceLastGuess > 0;
+    const hasPassedPixelsPerClassification = this.drawnPixelsSinceLastGuess >= this.config.pixelsPerClassify;
+    const hasPassedDefaultTimeBetweenClassify = this.timeSinceLastGuess >= this.config.defaultTimeBetweenClassify;
+    return (
+      hasPassedTimeToStartClassify &&
+      hasPassedMinimumTimeBetweenClassify &&
+      hasPassedMinimumDrawnThreshold &&
+      hasDrawnSinceLastClassify &&
+      !this.isDrawing &&
```
Comment (ComputasAlex) [inline src/app/game/game-draw/game-draw.component.ts]: Really like the logic, but personally not a big fan of !this.isDrawing. Found myself holding the mouse down for a long time drawing a palm, and it of course did not guess, but got correct immediately after releasing the mouse. Imagine many people hold down the finger for a long time. Should discuss with team and/or test on someone

File: src/app/admin/admin.component.ts [hunk 2]
```diff
@@ -11,7 +11,11 @@ import { AuthStatus } from '../shared/models/backend-interfaces';
   standalone: true,
 })
 export class AdminComponent {
-  constructor(private router: Router, private loginService: LoginService, private _snackBar: MatSnackBar) {}
+  constructor(
+    private router: Router,
```
Comment (ComputasAlex) [inline src/app/admin/admin.component.ts]: Different format configurations😩😩😩

File: src/app/game/services/game-config.service.ts [hunk 3]
```diff
@@ -6,6 +6,10 @@ export interface GameLevelConfig {
   rounds: number;
   secondsPerRound: number;
   timeToStartClassify: number;
+  defaultTimeBetweenClassify: number;
+  minimumTimeBetweenClassify: number;
```
Comment (ComputasAlex) [inline src/app/game/services/game-config.service.ts:10]: To deal with the never classifying of holding finger down for a long time, could consider introducing a maximumTimeBetweenClassify? So it at least classifies every 6 seconds or something?

**Review by ComputasAlex — APPROVED** [review]
Really like the logic, especially if it actually saves costs. 
Made a comment about the isDrawing, which I did not like while playing. Think we should discuss with the team or test on someone using an iPad.

File: src/app/admin/admin.component.ts (hunk 2 above)
Comment (andreasrberg) [reply src/app/admin/admin.component.ts]: think its fixed

**Review by ComputasAlex — APPROVED** [review]
Much better!

**Pull request merged**